include/streamable_flatbuffer.h

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#pragma once
+
+#include <iostream>
+#include <memory>
+#include <type_traits>
+
+#include "../fbs/flatbuffers/flatbuffers.h"
+
+namespace torchcraft {
+
+  // In/OutStreamableFlatBuffer
+  // Wrapper classes to Let us read/write Flatbuffer types to/from a stream.

```

Comment:
What's the advantage of having classes (with a single member function) instead of providing two templated functions?

Reply:
Attempting to not offending resident design sensibilities :D I prefer standalone functions myself. Will switch.